exercises/practice/queen-attack/queen_attack_test.go

```diff
@@ -1,51 +1,62 @@
-	w, b   string
-	attack bool
-	ok     bool
-}{
-	{"b4", "b4", false, false}, // same square
-	{"a8", "b9", false, false}, // off board
-	{"a0", "b1", false, false},
-	{"g3", "i5", false, false},
-	{"here", "there", false, false}, // invalid
-	{"", "", false, false},
-
-	{"b3", "d7", false, true}, // no attack
-	{"a1", "f8", false, true},
-	{"b4", "b7", true, true}, // same file
-	{"e4", "b4", true, true}, // same rank
-	{"a1", "f6", true, true}, // common diagonals
-	{"a6", "b7", true, true},
-	{"d1", "f3", true, true},
-	{"f1", "a6", true, true},
-	{"a1", "h8", true, true},
-	{"a8", "h1", true, true},
+type testCase struct {
+	description string
+	pos1, pos2  string
+	expected    bool
 }
 
-func TestCanQueenAttack(t *testing.T) {
-	for _, test := range tests {
-		switch attack, err := CanQueenAttack(test.w, test.b); {
-		case err != nil:
-			var _ error = err
-			if test.ok {
-				t.Fatalf("CanQueenAttack(%s, %s) returned error %q.  "+
-					"Error not expected.",
-					test.w, test.b, err)
+var validTestCases = []testCase{
+	{description: "no attack", pos1: "b3", pos2: "d7", expected: false},
+	{description: "no attack", pos1: "a1", pos2: "f8", expected: false},
+	{description: "same file", pos1: "b4", pos2: "b7", expected: true},
+	{description: "same rank", pos1: "e4", pos2: "b4", expected: true},
+	{description: "common diagonals", pos1: "a1", pos2: "f6", expected: true},
+	{description: "common diagonals", pos1: "a6", pos2: "b7", expected: true},
+	{description: "common diagonals", pos1: "d1", pos2: "f3", expected: true},
+	{description: "common diagonals", pos1: "f1", pos2: "a6", expected: true},
+	{description: "common diagonals", pos1: "a1", pos2: "h8", expected: true},
+	{description: "common diagonals", pos1: "a8", pos2: "h1", expected: true},
+}
+
+func TestCanQueenAttackValid(t *testing.T) {
+	for _, tc := range validTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			got, err := CanQueenAttack(tc.pos1, tc.pos2)
+			if err != nil {
+				t.Fatalf("CanQueenAttack(%q, %q) returned unexpected error %v", tc.pos1, tc.pos2, err)
 			}
-		case !test.ok:
-			t.Fatalf("CanQueenAttack(%s, %s) = %t, %v.  Expected error.",
-				test.w, test.b, attack, err)
-		case attack != test.attack:
-			t.Fatalf("CanQueenAttack(%s, %s) = %t, want %t.",
-				test.w, test.b, attack, test.attack)
-		}
+			if got != tc.expected {
+				t.Fatalf("CanQueenAttack(%q, %q) = %v, want: %v", tc.pos1, tc.pos2, got, tc.expected)
+			}
+		})
+	}
+}
+
+var invalidTestCases = []testCase{
+	{description: "same square", pos1: "b4", pos2: "b4", expected: false},
+	{description: "position off board", pos1: "a8", pos2: "b9", expected: false},
+	{description: "position off board", pos1: "a0", pos2: "b1", expected: false},
+	{description: "position off board", pos1: "g3", pos2: "i5", expected: false},
+	{description: "invalid position", pos1: "here", pos2: "there", expected: false},
+	{description: "empty position", pos1: "", pos2: "", expected: false},
```

Comment:
Since "expected" is always false for invalid cases (and we correctly do not even check it), I don't think it has to be listed in the test table.